hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HiveHoodieReaderContext.java

```diff
@@ -264,12 +270,27 @@ public UnaryOperator<ArrayWritable> projectRecord(Schema from, Schema to, Map<St
   }
 
   @Override
-  public Comparable castValue(Comparable value, Schema.Type newType) {
-    //TODO: [HUDI-8261] actually do casting here
+  public Comparable convertValueToEngineType(Comparable value) {
     if (value instanceof WritableComparable) {
       return value;
     }
-    return (WritableComparable) HoodieRealtimeRecordReaderUtils.avroToArrayWritable(value, Schema.create(newType));
+    //TODO: [HUDI-8261] cover more types
+    if (value == null) {
+      return null;
+    } else if (value instanceof String) {
+      return new Text((String) value);
+    } else if (value instanceof Integer) {
+      return new IntWritable((int) value);
+    } else if (value instanceof Long) {
+      return new LongWritable((long) value);
+    } else if (value instanceof Float) {
+      return new FloatWritable((float) value);
+    } else if (value instanceof Double) {
+      return new DoubleWritable((double) value);
+    } else if (value instanceof Boolean) {
```

Comment:
I don't even know if these datatypes will work for precombine only. since default value is picked as 0. 
can we test it out and updates docs on known limitations if applicable